File: experiments/bakeoff/benchmarks/python-inventory/ground_truth/reference/shop/types/inventory.py
"""types/inventory.py — Stock level + reservation token.

`StockLevel` tracks `on_hand` and `reserved` for a single sku.
`available == on_hand - reserved`. `ReservationToken` ties a stock
hold to an order_id; `committed` flips when the reservation is
applied to on_hand at ship time.
"""

from dataclasses import dataclass


@dataclass
class StockLevel:
    sku: str
    on_hand: int = 0
    reserved: int = 0

    @property
    def available(self) -> int:
        return self.on_hand - self.reserved


@dataclass
class ReservationToken:
    token_id: str
    order_id: str
    sku: str
    quantity: int
    committed: bool = False
    released: bool = False

    @property
    def is_open(self) -> bool:
        return not self.committed and not self.released

File: experiments/bakeoff/benchmarks/dart-inventory/ground_truth/reference/lib/types/inventory.dart
/// types/inventory.dart — Stock level + reservation token.
///
/// `StockLevel` tracks `onHand` and `reserved` for a single sku.
/// `available == onHand - reserved`. `ReservationToken` ties a stock
/// hold to an orderId; `committed` flips when the reservation is
/// applied to onHand at ship time.

class StockLevel {
  final String sku;
  int onHand;
  int reserved;

  StockLevel({required this.sku, this.onHand = 0, this.reserved = 0});

  int get available => onHand - reserved;
}

class ReservationToken {
  final String tokenId;
  final String orderId;
  final String sku;
  final int quantity;
  bool committed;
  bool released;

  ReservationToken({
    required this.tokenId,
    required this.orderId,
    required this.sku,
    required this.quantity,
    this.committed = false,
    this.released = false,
  });

  bool get isOpen => !committed && !released;
}
